### PR TITLE
Dynamic use of nagios-service checks

### LINKED ISSUE #1037: Nagios DNS Monitor Application for DNS records
In a dual-stacked IPv4 and IPv6 world dns hostnames are critical more then ever in a proper NMS environment. Checking the result of a returned dns query can be the difference between a successful NMS system and chasing your tail as to why services are failing in the NMS and the realworld.

Sometimes you simply have a dns record that you need to ensure exists or your environment might as well be offline to the world even if IP connectivity is fine.

Having the ability for your NMS to query a specific name server for a specific record and then checking that record against a defined result opens up many NMS options.

1. If you want to ensure that you company website is returning the primary IP. This would allow you check the dns query result to decided if everything is healthy or if you failed over to a DR site.

2. If you want to monitor a critical hostname for a specific IP. If another team member thinks they know better and modifies a domain name record for say an Email cluster. By default your MX record should return your mail clusters smtp HA/VIP IP to load-balance smtp connections. The employee updates the record to a specific IP. You can be alerted that this critical record has changed and take action. Maybe the change was needed during an emergency to restore service to one working box while you fix the other box's. Knowing this happened and having a visual alert to remind you to fix the dns record is helpful.

3. If you are running a Split DNS environment with External and Internal networks with both IPv4 and Ipv6 and over-lapping hostnames so humans don't need to remember two different hostnames based upon the network they are connected to. You need the ability to query a specific DNS server for a specific record to ensure both External and Internal environments are healthy.

4. If you are running a dual-stacked environment you need to confirm that a hostname returns the proper IP for both an A record and AAAA record.

http://wiki.zenoss.org/ZenPack:DNS_Monitor

This feature can leverage the check_dig Nagios plugin

### LINKED ISSUE #1792: Include all nagios-plugins
I'm trying to move our companies notifications server from Icinga to LibreNMS. The issue I am having though is that I followed the documentation to a T on adding the services and then nagios services to config.php and added what was said to add to cron.d file but I only see 16 services to use

config.php file
$config['show_services']           = 1;
$config['nagios_plugins']   = "/usr/lib64/nagios/plugins/";

Screen of services menu
http://i.imgur.com/JL4XHXc.png

As you can see below, there are far more services than what this menu shows

http://i.imgur.com/xOxqXi8.png

My goal is to essentially be able to use these plugins on LibreNMS

### PR DESCRIPTION
**Pull request by laf**
Fix #1792 
Fix #1037

This is a stab at providing this functionality. I added a service check for check_dig which doesn't exist as a current service check for us and this works fine.

It does make two assumptions:

1. The file is called check_*
2. That -H is required and passes the IP / hostname over.

Tidied up the table showing the checks and removed the broken graphs.

Also added the missing data for dbInsert to fix mysql strict bombing out.

### CODE REVIEW

**Comment by bmnnt** [comment]
ping

**Comment by f0o** [comment]
waiting for CI to come back

**Comment by laf** [comment]
Bump

**Comment by f0o** [comment]
Trying to do a simple ping check returns:
```
<wrta> was not set
check_ping: Could not parse arguments
Usage:
check_ping -H <host_address> -w <wrta>,<wpl>% -c <crta>,<cpl>%
 [-p packets] [-t timeout] [-4|-6]
```

Perhaps we should provide warning/critical parameters as dummies for now or write big notice-banners that the user needs to contribute them via the params form

**Comment by f0o** [comment]
Also small Change-Request:
- Hide the details-page button/link
- Order the checks alphabetically

I like the approach, would merge if not for the warning/critical params

File: html/pages/addsrv.inc.php [hunk 1]
```diff
@@ -18,10 +18,11 @@
         }
     }
 
-    if ($handle = opendir($config['install_dir'].'/includes/services/')) {
+    if ($handle = opendir($config['nagios_plugins'])) {
         while (false !== ($file = readdir($handle))) {
-            if ($file != '.' && $file != '..' && !strstr($file, '.')) {
-                $servicesform .= "<option value='$file'>$file</option>";
+            if ($file != '.' && $file != '..' && !strstr($file, '.') && strstr($file, 'check_')) {
+                list(,$check_name) = explode('_',$file);
```
Comment (f0o) [inline html/pages/addsrv.inc.php]: Fix `list(,$check_name) = explode('_',$file);` to `list(,$check_name) = explode('_',$file,2);`. Avoids cutoff where the plugin is called `check_rta_multi` etc.

**Comment by laf** [comment]
Added a small note about parameters may be needed.

Added ,2 to the explode.

Checks are already sorted by hostname.

Hide the details link.

**Pull request merged**